js/mixins/truncateTextMixin.js

```diff
@@ -0,0 +1,12 @@
+var React = require('react');
+var Utils = require('../components/utils');
+
+var TruncateTextMixin = {
+  truncateText: function(node, text) {
+    // CSS doesn't support "truncate N lines" so we need to do DOM width
+    // calculations to figure out where to truncate the description
+    var descriptionNode = React.findDOMNode(node);
+    descriptionNode.innerHTML = Utils.truncateTextToWidth(descriptionNode, text);
+  }
+};
+module.exports = TruncateTextMixin;
```

Comment:
Hi @chibzu, mainly to make it consistent across components.

Some components were using deprecated getDOMNode() or truncating text after the component mounted with state variables, it was easier to consolidate into one function.

And we may need to use this in additional components like discovery